src/api/app/controllers/trigger_controller.rb

```diff
@@ -66,10 +67,12 @@ def pundit_user
 
   def set_project_name
     @project_name = params[:project]
+    raise InvalidParameterError if @project_name.present? && @token.package && @token.package.project.name != @project_name
   end
 
   def set_package_name
     @package_name = params[:package]
-    raise MissingPackage if @package_name.blank? && @token.package_id.nil? && @token.token_name.in?(['rebuild', 'release', 'service'])
+    raise InvalidParameterError if @package_name.present? && @token.package && @token.package.name != @project_name
+    raise MissingPackage if @package_name.blank? && @token.package_id.nil? && @token.token_name.in?(['service'])
```

Comment:
What is the reason behind removing `release` and `rebuild` in the `in?(...)` method call?

Reply:
because the purpose here is to implement rebuild for projects, so bug #12813 was not fixing it, it was forbidding the feature from being used.

Reply:
> it was forbidding the feature from being used.

This is false.

Before, whenever a rebuild token without a package was triggered to a rebuild a project, the endpoint crashed with a 500. Passing a package to the endpoint in the query string would make it work. So whenever a package wasn't passed, the change recently introduced made the endpoint respond with a 400 instead of a 500. The behavior I am describing here is also what was already included in the API documentation.

![api-documentation](https://github.com/openSUSE/open-build-service/assets/1102934/a953c216-0175-404b-a562-f934cc203e80)

Reply:
the endpoint just crashed because of the introduced concern code. It should have worked before.

Reply:
> > it was forbidding the feature from being used.
> This is false.

> Before, whenever a rebuild token without a package was triggered to a rebuild a project, the endpoint crashed with a 500.

Yes, previously it was failing due to a bug, now its forbidding to use that. Thats what I said. 


> Passing a package to the endpoint in the query string would make it work.

we don't want to pass a package, we want to pass a project. There is a 7 year old feature request for it: #2094 

> So the feature can change, sure, but this gives you the reasoning behind the previous change.

You asked why the code got removed, not me ;)